src/Interface/Application/StateViewer.cc

```diff
@@ -0,0 +1,121 @@
+/*
+   For more information, please see: http://software.sci.utah.edu
+
+   The MIT License
+
+   Copyright (c) 2020 Scientific Computing and Imaging Institute,
+   University of Utah.
+
+   Permission is hereby granted, free of charge, to any person obtaining a
+   copy of this software and associated documentation files (the "Software"),
+   to deal in the Software without restriction, including without limitation
+   the rights to use, copy, modify, merge, publish, distribute, sublicense,
+   and/or sell copies of the Software, and to permit persons to whom the
+   Software is furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included
+   in all copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+   OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+   THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+   FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+   DEALINGS IN THE SOFTWARE.
+*/
+
+
+#include <QtGui>
+#include <QMenu>
+#include <Interface/Application/StateViewer.h>
+#include <Interface/Application/NetworkEditor.h>
+#include <Interface/Application/ModuleProxyWidget.h>
+#include <Interface/Application/ModuleWidget.h>
+
+using namespace SCIRun::Gui;
+using namespace SCIRun::Core;
+
+static const std::map<int, QString> variableTypeLabels =
```

Comment:
I guess my only question is why this doesn't belong privately to the class? Will a future class or something have access to this in the future?

Reply:
Probably not. But it's already static/private to the translation unit. A private member would be declared in the header and expose it even more.

Reply:
fair enough. Alright, I will approve!